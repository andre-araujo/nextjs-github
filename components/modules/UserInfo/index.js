import Avatar from '../../elements/Avatar';
import InfoListItem from '../../elements/InfoListItem';

const UserInfo = ({
    data,
}) => (
    <section>
        <Avatar url={data.avatar_url} />

        <ul>
            <InfoListItem
                title="Username:"
                description={data.login}
                url={data.htmlUrl}
            />

            <InfoListItem
                title="Followers:"
                description={data.followers}
                url={data.followers_url}
            />

            <InfoListItem
                title="Following:"
                description={data.following}
                url={data.following_url}
            />

            <InfoListItem
                title="Bio:"
                description={data.bio}
            />

            <InfoListItem
                title="email:"
                description={data.email}
            />
        </ul>
    </section>
);

export default UserInfo;
