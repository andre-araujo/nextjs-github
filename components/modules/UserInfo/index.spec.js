import React from 'react';
import { mount } from 'enzyme';

import UserInfo from './index';

describe('UserInfo component', () => {
    it('should match snapshot with all user data', () => {
        expect(mount(
            <UserInfo
                data={{
                    avatar_url: 'avatar_url',
                    login: 'login',
                    html_url: 'html_url',
                    followers: 34,
                    following: 23,
                    bio: 'bio',
                    email: 'email',
                }}
            />,
        )).toMatchSnapshot();
    });
});
