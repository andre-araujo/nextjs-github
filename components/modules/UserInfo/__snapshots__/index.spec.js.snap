// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component should match snapshot with all user data 1`] = `
<section>
  <div
    className="sc-bxivhb dhUpOj"
  >
    <img
      alt="user avatar"
      className="sc-ifAKCX eyjnnY"
      src="avatar_url"
    />
  </div>
  <div
    className="sc-bdVaJa csdvoq"
    is={null}
  >
    <ul>
      <li
        className="sc-EHOje kLGaUO"
      >
        <h3
          className="sc-bZQynM eycCXF"
        >
          Username:
        </h3>
        <p
          className="sc-gzVnrw hoHArn"
        >
          <a
            href="html_url"
            target="_blank"
          >
            login
          </a>
        </p>
      </li>
      <li
        className="sc-EHOje kLGaUO"
      >
        <h3
          className="sc-bZQynM eycCXF"
        >
          Followers:
        </h3>
        <p
          className="sc-gzVnrw hoHArn"
        >
          34
        </p>
      </li>
      <li
        className="sc-EHOje kLGaUO"
      >
        <h3
          className="sc-bZQynM eycCXF"
        >
          Following:
        </h3>
        <p
          className="sc-gzVnrw hoHArn"
        >
          23
        </p>
      </li>
      <li
        className="sc-EHOje kLGaUO"
      >
        <h3
          className="sc-bZQynM eycCXF"
        >
          Bio:
        </h3>
        <p
          className="sc-gzVnrw hoHArn"
        >
          bio
        </p>
      </li>
      <li
        className="sc-EHOje kLGaUO"
      >
        <h3
          className="sc-bZQynM eycCXF"
        >
          Email:
        </h3>
        <p
          className="sc-gzVnrw hoHArn"
        >
          email
        </p>
      </li>
    </ul>
  </div>
</section>
`;
